onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperator.cpp

```diff
@@ -111,6 +111,105 @@ namespace Dml
         }
     }
 
+    void DmlOperator::SetDmlOperatorGraphDesc(
+        const MLOperatorGraphDesc&& operatorGraphDesc,
+        const MLOperatorKernelCreationContext& kernelInfo
+        )
+    {
+        // Initialize should only be called once.
+        assert(m_compiledOperator == nullptr);
+
+        // DML doesn't support empty tensors. If an operator is still executable with empty tensors, the empty tensors
+        // should be removed or massaged depending on the definition.
+        for (const TensorDesc& desc : m_inputTensorDescs)
+        {
+            if (OperatorHelper::ContainsEmptyDimensions(desc.GetSizes()))
+            {
+                return;
+            }
+        }
+
+        for (const TensorDesc& desc : m_outputTensorDescs)
+        {
+            if (OperatorHelper::ContainsEmptyDimensions(desc.GetSizes()))
+            {
+                return;
+            }
+        }
+
+        // graphInputIndices and m_kernelInputIndices should be identity
+        for (uint32_t idx = 0; idx < operatorGraphDesc.inputEdgeCount; idx++)
+        {
+            if (m_kernelInputIndices[idx] == std::nullopt || !kernelInfo.IsInputValid(*m_kernelInputIndices[idx]))
+            {
+                continue;
+            }
+            assert(m_kernelInputIndices[idx] == operatorGraphDesc.inputEdges[idx].GraphInputIndex);
+        }
+
+        // graphOutputIndices and m_kernelOutputIndices should be identity
+        for (uint32_t idx = 0; idx < operatorGraphDesc.outputEdgeCount; idx++)
+        {
+            if (m_kernelOutputIndices[idx] == std::nullopt || !kernelInfo.IsOutputValid(*m_kernelOutputIndices[idx]))
+            {
+                continue;
+            }
+            assert(m_kernelOutputIndices[idx] == operatorGraphDesc.outputEdges[idx].GraphOutputIndex);
```

Comment:
This should assert m_kernelInputIndices[idx] == idx.  There is not necessarily an association between graph edge index and operator input index. E.g., there could be two graph edges going from same operator input to different internal graph nodes